### PR TITLE
move references from TOOL_BRANCH to be TOOL_REF

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

move references from TOOL_BRANCH to be TOOL_REF

to be honest I'm not 100% sure if this is enough, opening the PR for feedback

/assign @saschagrunert @hasheddan @puerco 

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/release/issues/1028

#### Does this PR introduce a user-facing change?

```release-note
move references from TOOL_BRANCH to be TOOL_REF
```

### CODE REVIEW

File: gcb/release/cloudbuild.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ steps:
   dir: "go/src/k8s.io"
   args:
   - "clone"
-  - "--branch=${_TOOL_BRANCH}"
+  - "--branch=${_TOOL_REF}"
```
Comment (puerco) [inline gcb/release/cloudbuild.yaml:24]: As it stands we will successfully get the value from the substitution here. But I think that this still leaves us only with the possibility to use a specific branch (or a tag). We still need to check out a commit if we want to use an arbitrary ref.

File: gcb/release/cloudbuild.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ steps:
...
   - "https://github.com/${_TOOL_ORG}/${_TOOL_REPO}"
```
Comment (puerco) [inline gcb/release/cloudbuild.yaml:26]: One possibility to checkout a ref would be to insert a new GCB stage here that simply does `git checkout SHA` when TOOL_REF is a string that looks like a sha or sha+branch slug. It should be fairly quick as the git container is already cached and is one of the official GCB builders anyway.

The other option would be to change the clone stage, modify the entry point to sh/bash, and call git twice from the shell (clone and then checkout).

**Review by puerco — COMMENTED** [review]
I think we still need to add a bit of plumbing here to checkout a specific commit if we want to use a specific ref instead of a branch, let me know what you think:

File: gcb/release/cloudbuild.yaml (hunk 2 above)
Comment (cpanato) [reply gcb/release/cloudbuild.yaml:26]: thanks for the feedback, I was unsure about that, I will ping you in slack to discuss if you can offcourse

**Comment by cpanato** [comment]
when have some free cycles can you take a look at this change? thanks! 
@saschagrunert @puerco

File: pkg/git/git.go [hunk 3]
```diff
@@ -459,7 +460,31 @@ func (r *Repo) RevParse(rev string) (string, error) {
...
+// error.
+func (r *Repo) RevParse(rev string) (string, error) {
+	// Try to resolve the rev
+	ref, err := r.inner.ResolveRevision(plumbing.Revision(rev))
+	if err != nil {
+		return "", err
+	}
+
+	return ref.String(), nil
+}
```
Comment (puerco) [inline pkg/git/git.go:473]: I'm amazed at how much work you put in to rename the old `git.RevParse()`. I think I would have added the functionality to the same function ;)

**Review by puerco — COMMENTED** [review]
In general, it looks good to me. Renaming `git.RevParse()` touches a lot of things but I'm confident you got them all covered. From my perspective it should work, I would vote to merge it after 1.21 beta.0 and give it a few test runs.

File: pkg/git/git.go [hunk 4]
```diff
@@ -459,7 +460,31 @@ func (r *Repo) RevParse(rev string) (string, error) {
 	return ref.String(), nil
 }
 
-// RevParseShort parses a git revision and returns a SHA1 trimmed to the length
+// RevParseTag parses a git revision and returns a SHA1 on success, otherwise an
```
[the remark this answers is not in the document]
Comment (cpanato) [reply pkg/git/git.go:463]: done, thanks for the review <3

File: pkg/git/git.go [hunk 5]
```diff
@@ -459,7 +460,31 @@ func (r *Repo) RevParse(rev string) (string, error) {
 	return ref.String(), nil
```
Comment (saschagrunert) [inline pkg/git/git.go:460]: Let's just call `RevParse` here.

**Review by saschagrunert — COMMENTED** [review]
Code LGTM beside the single nit. @cpanato can you give it some manual test runs?

**Comment by cpanato** [comment]
using:
TOOL_ORG=cpanato
TOOL_REF=91a1579eb3161d6c805a77e01e89893baa3a9885

from here: https://github.com/cpanato/release/commit/91a1579eb3161d6c805a77e01e89893baa3a9885

stage build: https://console.cloud.google.com/cloud-build/builds/f3ede851-254b-4ad7-bd26-2aa3349ab3b3?project=kubernetes-release-test

and in the logs can see the custom change
```
Step #3: INFO[2021-03-21T09:41:59Z] Using CUSTOM ORG/REPO/BRANCH:
Step #3: cpanato/release/91a1579eb3161d6c805a77e01e89893baa3a9885 
```

cc @saschagrunert @puerco

**Comment by cpanato** [comment]
using:
TOOL_ORG=cpanato
TOOL_REF=testing-ref

from here: https://github.com/cpanato/release/tree/testing-ref

stage build: https://console.cloud.google.com/cloud-build/builds/e312f9a4-c279-4944-b24d-71ff44ba40d6?project=kubernetes-release-test

and in the logs can see the custom change
```
Step #3: INFO[2021-03-21T11:57:16Z] Using CUSTOM ORG/REPO/BRANCH:
Step #3: cpanato/release/testing-ref 
```

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**